app/api/resources/user.py

```diff
@@ -498,3 +498,104 @@ def get(cls):
             return messages.USER_NOT_FOUND, HTTPStatus.NOT_FOUND
 
         return dashboard, HTTPStatus.OK
+
+@users_ns.route("user/forgot_password")
+class ForgotPassword(Resource):
+    @classmethod
+    @users_ns.doc("forgot password")
+    @users_ns.response(200, "%s" % messages.PASSWORD_RESET_MAIL_MESSAGE)
+    @users_ns.response(400, "%s" % messages.EMAIL_FIELD_IS_MISSING)
+    @users_ns.response(403, "%s" % messages.USER_HAS_NOT_VERIFIED_EMAIL_BEFORE_LOGIN)
+    @users_ns.response(404, "%s" % messages.USER_IS_NOT_REGISTERED_IN_THE_SYSTEM)
+    @users_ns.expect(forgot_password_change_request_data_model, validate=True)
+ 
+    def post(self):
+        """ Updates User's password by sending reset link to email
+ 
+        Argument: 
+          email : Verified email of the user
+ 
+        Return:
+          token: a token link/reset link will be shared on verified mail of user
+        """
+ 
+        data = request.json
+ 
+        if not data:
+            return messages.EMAIL_FIELD_IS_MISSING, 400
+ 
+        is_valid = validate_forgot_password_email(data)
+ 
+        if is_valid!={}:
+            return is_valid, 400
+        
+        user = DAO.get_user_by_email(data["email"])
+ 
+        if user is None:
+            return messages.USER_IS_NOT_REGISTERED_IN_THE_SYSTEM, 404
+ 
+        if not user.is_email_verified:
+            return messages.USER_HAS_NOT_VERIFIED_EMAIL_BEFORE_LOGIN, 403    
+ 
+        send_email_reset_password_message(user.name, data["email"])
+ 
+        return messages.PASSWORD_RESET_MAIL_MESSAGE, 200
+ 
+@users_ns.route("user/reset_password")
+@users_ns.response(201, "%s" % messages.PASSWORD_SUCCESSFULLY_UPDATED)
```

Comment:
201 is used when a new resource is created, does this endpoint do that? @robotjellyzone 
Here's a list of status codes - https://httpstatuses.com/ - I am thinking of using 200 🤔

Reply:
So, should i use 205 or 200 as 205 is for **reset content** and here we are reseting our password also, i think 200 can also be fine so, which one should i use @isabelcosta

Reply:
Ok used 200 ie **HttpStatus.OK** :)